scripts/o3de/o3de/engine_template.py

```diff
@@ -1815,6 +1815,17 @@ def create_gem(gem_path: pathlib.Path,
                template_path: pathlib.Path = None,
                template_name: str = None,
                gem_name: str = None,
+               display_name: str = None,
```

Comment:
We really need to split this file up based on the commands exposed(`create-template`, `create-from-template`, `create-gem`, `create-project`).

A lot of this logic is duplicate in each those methods.

Reply:
I agree, though I don't think I'll get to this change in this PR - just trying to help unblock the Create Gem feature